lib/rules/color-named/index.js

```diff
@@ -58,11 +58,23 @@ const rule = function (expectation, options) {
         return
       }
 
+      const functionNodes = []
+
       valueParser(decl.value).walk(node => {
         const value = node.value,
           type = node.type,
           sourceIndex = node.sourceIndex
 
+        if (type === "function") {
```

Comment:
Yes, we definitely need a clarifying comment here. Also, I think the variable name `functionNodes` is misleading. You're not actually putting *nodes* in it, right? Could you please try to name that array to match its contents?